ui/app/components/secure-variable-form/namespace-filter.js

```diff
@@ -0,0 +1,26 @@
+// @ts-check
+
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+import Component from '@glimmer/component';
+
+export default class NamespaceFilter extends Component {
+  @service store;
+
+  @action
+  async fetchNamespaces() {
+    return this.store.findAll('namespace');
+  }
+
+  @action
+  formatAndSetNamespaces() {
+    // Triggered on the promise in fetchNamespaces resolving
+    const namespaces = this.store
+      .peekAll('namespace')
+      .map(({ name }) => ({ key: name, label: name }));
+
+    if (namespaces.length <= 1) return null;
```

Comment:
Just want to check on this: are we sure the logic here is okay?

What if I have access to a single namespace (`my-namespace`), but not `*` / `default` ?

Will this mean that, when trying to create a new variable, it doesn't automatically set my namespace to `my-namespace`, and instead falls back to `*`, which would fail on POST?

Reply:
`formatAndSetNamespaces` is for generating the namespaces that will appear in the dropdown list. First, we peek in the store to grab a list of the namespaces. So, if you only have access to `my-namespace` then we would presume that the API returns only `my-namespace`.

The `/namespaces` endpoint is just going to return a list of namespaces -- we're not responsible for knowing if the API is returning results that the user should not have access to.

Side Note:  A user should not be able to set `*` as a namespace. We use `*` as a wildcard for searching and filtering namespaces in our `get` requests.

Reply:
Right! The `Namespace.List` RPC that populates that response returns only namespaces you have access to (ref  [`namespace_endpoint.go#L252-L255`](https://github.com/hashicorp/nomad/blob/v1.3.2/nomad/namespace_endpoint.go#L252-L255)), even if you pass `*` wildcard in the request.